src/Symfony/Component/Validator/Constraints/ImageValidator.php

```diff
@@ -43,6 +44,22 @@ public function validate($value, Constraint $constraint)
             return;
         }
 
+        if ($constraint->strict) {
+            if (!function_exists('imagecreatefromstring')) {
+                throw new RuntimeException('Strict image validation requires installed and enabled GD extension');
+            }
+
+            $resource = @imagecreatefromstring(file_get_contents($value));
```

Comment:
2 things: first, depending on the value (which is user submitted), the resource can be huge in terms of memory and then, this is an attack vector depending on possible exploits found in image file formats. I think this should be not included in Symfony core for these reasons.